readme.txt
Developer: Crisilyn Paguio

Application Title: Fun With Math

Application Overview:
  This application is an educational Math game for kids in primary level. It has
  three levels, and each level has four mini games to be completed in order to
  get a reward, which will help a farmer build his own farm.

Major Features:
  This game includes the following features:
  -Four fun mini games to make learning Math more exciting
  -Different levels with different difficulties to cater to different levels of math knowledge
  -Rewards after the completion of each mini game
  -Multiple chances to get the correct answer
  -No time limit to give children time to solve for the math problems
  -Positive feedbacks for encouragement

Major functions
  -displayHomePage: displays the home page of the app
  -chooseLevel: displays the levels available and the descriptions of each level
  -createEasyQAGame, createMediumQAGame, createHardQAGame: create the question and answer games (easy, medium, hard)
  -generateEasyQuestion, generateMediumQuestion, generateHardQuestion: generate the questions for the question and answer game for all levels (easy, medium, hard)
  -checkAnswerForEasyLevel, checkAnswerForMediumLevel, checkAnswerForHardLevel: check the answers for the question and answer game for all levels (easy, medium, hard)
  -createMemoryGame, createMediumMemoryGame, createHardMemoryGame: create the memory games (easy, medium, hard)
  -createEasyMultipleChoiceGame, createMediumMultipleChoiceGame, createHardMultipleChoiceGame: create the multiple choice games (easy, medium, hard)
  -generateMultipleChoiceQuestionAndOptions, generateMultipleChoiceMediumQuestionAndOptions, generateMultipleChoiceHardQuestionAndOptions: generate the questions and options for the multiple choice games for all levels (easy, medium, hard)
  -checkForMCAnswer, checkForMCMediumAnswer, checkForMCHardAnswer: check the answers for the multiple choice games for all levels (easy, medium, hard)
  -createMatchTheAnswerGame, createMatchTheAnswerGame2,createMatchTheAnswerGame3: create match the answer game for the easy level
  -createMatchTheAnswerMediumGame, createMatchTheAnswerMediumGame2, createMatchTheAnswerMediumGame3: create match the answer game for the medium level
  -createMatchTheAnswerHardGame, createMatchTheAnswerHardGame2, createMatchTheAnswerHardGame3: create match the answer game for the hard level
  -allowDrop, onNumberDragStart: functions for the drop and drag
  -createOnBoxDropFunctionEasy, createOnBoxDropFunctionEasy2, createOnBoxDropFunctionEasy3,
   createOnBoxDropFunctionMedium, createOnBoxDropFunctionMedium2, createOnBoxDropFunctionMedium3,
   createOnBoxDropFunctionHard, createOnBoxDropFunctionHard2, createOnBoxDropFunctionHard3: functions that are called on the container that receives a drag event
  -createTransition: displays a congratulatory message and a reward after the completion of each mini-game, except for the last.
  -endOfLevelPage: displays a congratulatory message and a reward. This also lets the user go back to the "Choose Level" page, go to "See Farm" page, or continue to the next level.
  -endOfGamePage: displays a congratulatory message and a reward. This also lets the user go to "See Farm" page or go back to the "Choose Level" page.
  -seeFarm, seeFarmEasy, seeFarmMedium: display the current state of the farm. This page shows the rewards that the user got from the mini-games completed.

  Data Structure:
    -Question-Answer game: array of objects; each object consists of a question and its answer
    -Memory game: array of objects; each object consists of a name and an image source
    -Multiple choice: array of objects; each object is consists of a number, image, and options, which consist of objects of options for the questions
    -Matching game: array of strings that consist of the math problems and the name of the numbers, which are answers to the math problems


changelog.txt
25/03/2018
  -created the project directory (folders and files)
  -added images

  Progress:
  Project has been set-up.


01/04/2018
  -started creating the 1st level of the game
  -created the functions for creating the ui of the app

  Progress:
  My project has now a mini-game and the functions to create the UI of the application.

15/04/2018
  -finished creating the 1st level of the game without the design yet

  Progress:
  My project has now one mini-game and ready to have more mini-games.

22/04/2018
  -added more questions to easy level
  -style(animation) is added to give visual feedback to user when the answer is correct/wrong

  Progress:
  The first mini-game lasts longer due to additional questions. Also, users are given visual feedback to their answers.


24/04/2018
  -added more images needed for the mini games

  -still working on completing the easy level

  Progress:
  The images needed to create my application are gathered.

26/04/2018
  -added a matching type mini game

  Progress:
  My application has now two mini-games, which will help fulfill its features of having mini-games.

28/04/2018
  -added a background image
  -added a transition page when a mini game ends
  -added styles
  -connected the 1st mini game to the 2nd mini game
  -created end-of-mini-game page

  -still working on the matching type game to make it compatible with mobile devices.
  -still working on adding more mini games

  Progress:
  My application is looking to be more like a real application due to the background image added, pages added, and transitions.

02/05/2018
  -still working on a solution for displaying the prizes the user gets after completing the mini-games

  Progress:
  Some bugs are fixed, and my application has now playable scenes.

04/05/2018
  -added responsive css
  -instead of just displaying the answer, the answer is being told by the pig character in the question-answer game

  -still working on improving the design of the mini-games
  -still working on making the matching game work on mobile phones
  -still working on adding more levels and mini-games

  Progress:
  -My app's design is now responsive to mobile devices resolution.


05/05/2018
  -added a memory game

  -still working on improving the design of the mini-games
  -still working on making the matching game work on mobile phones
  -still working on adding more levels and mini-games

  Progress:
  The playability of my app is improved by having another mini game.

08/05/2018
  -added a multiple question game

  -still working on making the memory game's flipping of cards work properly (images do not appear when clicked)
  -still working on making the matching game work on mobile phones
  -still working on adding more levels

  Progress:
  The playability of my app is improved by having another mini game.

09/05/2018
  -fixed the problem with flipping of cards in the memory game

  -still working on making the matching game work on mobile phones
  -still working on fixing the multiple choice mini-game (the question divs are overlapping when another question is loaded)
  -still working on adding more levels

  Progress:
  My app has now more playable scenes.

10/05/2018
  -fixed the problem with overlapping questions and options in the multiple choice mini-game

  -still working on making the matching game work on mobile phones
  -still working on adding more levels

  Progress:
  Bugs are fixed, which helped in the improvement of my app.

12/05/2018
  -randomized the generated questions in the question-answer mini-game

  -still working on making the matching game work on mobile phones
  -still working on adding more levels

  Progress:
  Questions in some of the mini-games are randomised.

13/05/2018
  -checked if all cards in the memory game matched
  -added transition from completed memory game to end of level Page
  -added a home Page
  -added a 'chooseLevel' page
  -fixed seeFarm page
  -added instructions to the mini-games
  -added medium and hard levels

  -still working on making the matching game work on mobile phones

  Progress:
  My project has now a real app feels due to the pages added.

14/05/2018
  -fixed the matching game
  -if-statement is added to check if all cards match in the memory game
  -completed all levels of all mini-games
  -added an animation

  -still working on styles
  -still working on adding more questions and animations


  Progress:
  My project is almost complete because of the completion of all pages.

15/05/2018
  -randomized question items in the matching type mini-games
  -added more questions to multiple choice mini-games and matching type mini-games (all levels)
  -completed all mini-games in all levels

  -still working on making the style responsive to other devices aside from mobile devices.


  Progress:
  My app has all functionalities planned and completed all pages.


16/05/2018
  -made the UI display responsive to web and small devices landscape mode

  -still working on making the style more responsive

    Progress:
    My app is now more responsive.

17/05/2018
  -made the UI display responsive to web, tablet, and small devices landscape mode

  Progress:
  My app is now responsive to three devices.


js/ui.js
/*
  This function creates a paragraph element.
  It accepts a text, which will be the content of the paragraph, and a class name that will be assigned to it.
  This function will be called on the parts of the game where text is needed such as instructions.
  It will return a paragraph element.
*/
function createParagraph(text, className){
  var p = document.createElement("P");

  //assign the classname if not assigned yet
  if (typeof className !== 'undefined'){
    p.classList.add(className);
  }
  document.body.appendChild(p);
  var t = document.createTextNode(text);
  p.appendChild(t);

  return p;
}

/*
  This function creates a h1 element.
  It accepts a text, which will be the content of the heading.
  This function will be called in creating the title of the game.
  It will return an h1 element.
*/
function createHeadings(headingText){
  var heading = document.createElement("H1");
  document.body.appendChild(heading);
  var t = document.createTextNode(headingText);
  heading.appendChild(t);

  return heading;
}

/*
  This function creates a div element.
  This function will be called whenever other html elements such as images and texts need a container.
  It will return a div element.
*/
function createDiv(){
  var d = document.createElement("DIV");
  document.body.appendChild(d);

  return d;
}


/*
  This function creates a button element.
  It accepts a text, which will be the text that will be displayed on the button.
  This function will be called in creating buttons for the game.
  It will return an button element.
*/
function createButton(buttonText){
  var b = document.createElement("BUTTON");
  var t = document.createTextNode(buttonText);
  b.appendChild(t);
  document.body.appendChild(b);

  return b;
}

/*
  This function creates a text input element.
  This function will be called in creating text inputs for the users.
  It will return a text input element.
*/
function createTextInput(){
  var tB = document.createElement("INPUT");
  document.body.appendChild(tB);
  tB.setAttribute("type", "text");

  return tB;
}

/*
  This function creates a radio element.
  It accepts a text, which will be the name of the radio.
  This function will be called in creating radio buttons for the options in the multiple choice game.
  It will return a radio element.
*/
function createRadioButtons(name){
  var rB = document.createElement("INPUT");
  document.body.appendChild(rB);
  rB.setAttribute("type", "radio");
  rB.setAttribute("name", name);

  return rB;
}


/*
  This function creates an image element.
  It accepts a text, which will be the source of the image file.
  This function will be called in creating the images of the game.
  It will return an image element.
*/
function createImage(source){
  var img = document.createElement("IMG");
  img.src = source;
  document.body.appendChild(img);

  return img;
}

/*
  This function creates a section element.
  This function will be called in creating the cards for the memory game.
  It will return a section element.
*/
function createSection(){
  var sec = document.createElement("SECTION");
  document.body.appendChild(sec);

  return sec;
}

/*
  This function creates a label element.
  This function will be called in creating labels for the radio buttons.
  It is used in creating the options for the multiple choice game.
  It will return a label element.
*/
function createLabel(){
  var label = document.createElement("LABEL");
  document.body.appendChild(label);

  return label;
}


data/easyLevel.js
/*
  This contains the data for the easy level.
*/

/* Array of numbers to be used in randomising questions */
window.numbers = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];


/* Question and answer game data */
window.easyLevel = [
  {
    question: "2 + 5 = ",
    answer: 7
  },
  {
    question: "3 + 1 = ",
    answer: 4
  },
  {
    question: "9 - 1 = ",
    answer: 8
  },
  {
    question: "2 - 1 = ",
    answer: 1
  },
  {
    question: "5 * 1 = ",
    answer: 5
  },
  {
    question: "6 * 0 = ",
    answer: 0
  },
  {
    question: "4 / 2 = ",
    answer: 2
  },
  {
    question: "8 / 2 = ",
    answer: 4
  },
  {
    question: "1 / 1 = ",
    answer: 1
  },
  {
    question: "6 / 3 = ",
    answer: 2
  }
];

/* Memory game data */
window.memoryGameElements = [
  {
    name: "one",
    image: "images/one.png"
  },
  {
    name: "two",
    image: "images/two.png"
  },
  {
    name: "three",
    image: "images/three.png"
  },
  {
    name: "four",
    image: "images/four.png"
  },
  {
    name: "five",
    image: "images/five.png"
  },
  {
    name: "six",
    image: "images/six.png"
  },
  {
    name: "seven",
    image: "images/seven.png"
  },
  {
    name: "eight",
    image: "images/eight.png"
  },
  {
    name: "nine",
    image: "images/nine.png"
  }

];


/* Multiple Choice game data */
window.multipleChoiceElements = [
  {
    number: 3,
    animalImage: "images/duck.png",
    options: {
      a: "2",
      b: "3",
      c: "5"
    },
    answer: "b"
  },
  {
    number: 1,
    animalImage: "images/goat.png",
    options: {
      a: "4",
      b: "3",
      c: "1"
    },
    answer: "c"
  },
  {
    number: 5,
    animalImage: "images/chicken.png",
    options: {
      a: "2",
      b: "4",
      c: "5"
    },
    answer: "c"
  },
  {
    number: 2,
    animalImage: "images/sheep.png",
    options: {
      a: "2",
      b: "6",
      c: "1"
    },
    answer: "a"
  },
  {
    number: 6,
    animalImage: "images/chick.png",
    options: {
      a: "6",
      b: "3",
      c: "5"
    },
    answer: "a"
  },
  {
    number: 4,
    animalImage: "images/cow.png",
    options: {
      a: "7",
      b: "4",
      c: "5"
    },
    answer: "b"
  },
  {
    number: 7,
    animalImage: "images/donkey.png",
    options: {
      a: "6",
      b: "7",
      c: "5"
    },
    answer: "b"
  },
  {
    number: 3,
    animalImage: "images/chicken2.png",
    options: {
      a: "4",
      b: "5",
      c: "3"
    },
    answer: "c"
  },
  {
    number: 5,
    animalImage: "images/goat2.png",
    options: {
      a: "5",
      b: "3",
      c: "6"
    },
    answer: "a"
  },
  {
    number: 2,
    animalImage: "images/egg.png",
    options: {
      a: "3",
      b: "2",
      c: "4"
    },
    answer: "b"
  }
];

/* Matching game data */
window.matchingGameProblems = [

    "9 - 6",
    "5 + 2",
    "9 - 8"
];

window.matchingGameElements = [

    "three",
    "seven",
    "one"
];

window.matchingGameProblems2 = [

    "5 + 4",
    "3 * 2",
    "8 / 2"
];

window.matchingGameElements2 = [

    "nine",
    "six",
    "four"
];

window.matchingGameProblems3 = [

    "8 * 1",
    "7 - 5",
    "4 + 1"
];

window.matchingGameElements3 = [

    "eight",
    "two",
    "five"
];


data/mediumLevel.js
/*
  This contains the data for the medium level.
*/

/* Question and answer data */
window.mediumLevel = [
  {
    question: "2 + 10 = ",
    answer: 12
  },
  {
    question: "5 + 8 = ",
    answer: 13
  },
  {
    question: "16 - 4 = ",
    answer: 12
  },
  {
    question: "9 - 6 = ",
    answer: 3
  },
  {
    question: "6 * 3 = ",
    answer: 18
  },
  {
    question: "2 * 5 = ",
    answer: 10
  },
  {
    question: "12 / 2 = ",
    answer: 6
  },
  {
    question: "15 / 5 = ",
    answer: 3
  },
  {
    question: "20 / 2 = ",
    answer: 10
  },
  {
    question: "18 / 6 = ",
    answer: 3
  }
];

/* Memory game data */
window.memoryGameMediumElements = [
  {
    name: "one",
    image: "images/raw/one.jpg"
  },
  {
    name: "two",
    image: "images/raw/two.jpg"
  },
  {
    name: "three",
    image: "images/raw/three.jpg"
  },
  {
    name: "four",
    image: "images/raw/four.jpg"
  },
  {
    name: "five",
    image: "images/raw/five.jpg"
  },
  {
    name: "six",
    image: "images/raw/six.jpg"
  },
  {
    name: "seven",
    image: "images/raw/seven.jpg"
  },
  {
    name: "eight",
    image: "images/raw/eight.jpg"
  },
  {
    name: "nine",
    image: "images/raw/nine.jpg"
  }

];

/* Multiple Choice game data */
window.multipleChoiceMediumElements = [
  {
    number: 9,
    animalImage: "images/donkey.png",
    options: {
      a: "8",
      b: "10",
      c: "9"
    },
    answer: "c"
  },
  {
    number: 7,
    animalImage: "images/pig.png",
    options: {
      a: "7",
      b: "8",
      c: "9"
    },
    answer: "a"
  },
  {
    number: 10,
    animalImage: "images/sheep.png",
    options: {
      a: "10",
      b: "9",
      c: "11"
    },
    answer: "a"
  },
  {
    number: 6,
    animalImage: "images/chicken2.png",
    options: {
      a: "6",
      b: "9",
      c: "5"
    },
    answer: "a"
  },
  {
    number: 11,
    animalImage: "images/egg.png",
    options: {
      a: "10",
      b: "11",
      c: "7"
    },
    answer: "b"
  },
  {
    number: 8,
    animalImage: "images/goat2.png",
    options: {
      a: "7",
      b: "8",
      c: "5"
    },
    answer: "b"
  },
  {
    number: 14,
    animalImage: "images/duck.png",
    options: {
      a: "14",
      b: "13",
      c: "15"
    },
    answer: "a"
  },
  {
    number: 12,
    animalImage: "images/chick.png",
    options: {
      a: "16",
      b: "13",
      c: "12"
    },
    answer: "c"
  },
  {
    number: 13,
    animalImage: "images/cow.png",
    options: {
      a: "16",
      b: "13",
      c: "15"
    },
    answer: "b"
  },
  {
    number: 9,
    animalImage: "images/chicken.png",
    options: {
      a: "6",
      b: "9",
      c: "8"
    },
    answer: "b"
  }
];

/* Matching game data */
window.matchingGameMediumProblems = [

    "35 - 26",
    "6 * 0",
    "40 / 8"
];

window.matchingGameMediumElements = [

    "nine",
    "zero",
    "five"
];

window.matchingGameMediumProblems2 = [

    "7 * 2",
    "12 + 5",
    "20 - 10"
];

window.matchingGameMediumElements2 = [

    "fourteen",
    "seventeen",
    "ten"
];

window.matchingGameMediumProblems3 = [

    "30 / 10",
    "19 + 5",
    "6 * 3"
];

window.matchingGameMediumElements3 = [

    "three",
    "twentyFour",
    "eighteen"
];


data/hardLevel.js
/*
  This contains the data for the hard level.
*/

/* Question and answer game data */
window.hardLevel = [
  {
    question: "10 * 10 = ",
    answer: 100
  },
  {
    question: "100 + 5 = ",
    answer: 105
  },
  {
    question: "34 + 10 = ",
    answer: 44
  },
  {
    question: "16 + 24 = ",
    answer: 40
  },
  {
    question: "73 - 19 = ",
    answer: 54
  },
  {
    question: "55 - 16 = ",
    answer: 39
  },
  {
    question: "17 - 22 = ",
    answer: -5
  },
  {
    question: "18 * 3 = ",
    answer: 54
  },
  {
    question: "96 / 12 = ",
    answer: 8
  },
  {
    question: "125 / 25 = ",
    answer: 5
  }
];

/* Memory game data */
window.memoryGameHardElements = [
  {
    name: "one",
    image: "images/one.png"
  },
  {
    name: "two",
    image: "images/two.png"
  },
  {
    name: "three",
    image: "images/three.png"
  },
  {
    name: "four",
    image: "images/four.png"
  },
  {
    name: "five",
    image: "images/five.png"
  },
  {
    name: "six",
    image: "images/six.png"
  },
  {
    name: "seven",
    image: "images/seven.png"
  },
  {
    name: "eight",
    image: "images/eight.png"
  },
  {
    name: "nine",
    image: "images/nine.png"
  }

];

window.memoryGameHardElements2 = [
  {
    name: "one",
    image: "images/raw/one.png"
  },
  {
    name: "two",
    image: "images/raw/two.png"
  },
  {
    name: "three",
    image: "images/raw/three.png"
  },
  {
    name: "four",
    image: "images/raw/four.png"
  },
  {
    name: "five",
    image: "images/raw/five.png"
  },
  {
    name: "six",
    image: "images/raw/six.png"
  },
  {
    name: "seven",
    image: "images/raw/seven.png"
  },
  {
    name: "eight",
    image: "images/raw/eight.png"
  },
  {
    name: "nine",
    image: "images/raw/nine.png"
  }

];

/* Multiple Choice game data */
window.multipleChoiceHardElements = [
  {
    questionImage: "images/raw/mc1.jpg",
    options: {
      a: "98",
      b: "81",
      c: "91"
    },
    answer: "b"
  },
  {
    questionImage: "images/raw/mc3.jpg",
    options: {
      a: "7",
      b: "8",
      c: "9"
    },
    answer: "c"
  },
  {
    questionImage: "images/raw/mc4.jpg",
    options: {
      a: "68",
      b: "71",
      c: "72"
    },
    answer: "c"
  },
  {
    questionImage: "images/raw/mc5.jpg",
    options: {
      a: "27",
      b: "26",
      c: "24"
    },
    answer: "b"
  },
  {
    questionImage: "images/raw/mc2.jpg",
    options: {
      a: "-26",
      b: "74",
      c: "26"
    },
    answer: "a"
  },
  {
    questionImage: "images/raw/mc6.jpg",
    options: {
      a: "70",
      b: "48",
      c: "51"
    },
    answer: "b"
  },
  {
    questionImage: "images/raw/mc7.jpg",
    options: {
      a: "10",
      b: "18",
      c: "11"
    },
    answer: "a"
  },
  {
    questionImage: "images/raw/mc8.jpg",
    options: {
      a: "56",
      b: "48",
      c: "51"
    },
    answer: "a"
  },
  {
    questionImage: "images/raw/mc9.jpg",
    options: {
      a: "70",
      b: "60",
      c: "51"
    },
    answer: "b"
  },
  {
    questionImage: "images/raw/mc10.jpg",
    options: {
      a: "60",
      b: "58",
      c: "57"
    },
    answer: "c"
  }
];

/* Matching type game data */
window.matchingGameHardProblems = [

    "74 - 56",
    "12 + 13",
    "18 + 21"
];

window.matchingGameHardElements = [

    "eighteen",
    "twentyFive",
    "thirtyNine"
];


window.matchingGameHardProblems2 = [

    "46 / 2",
    "9 * 11",
    "110 - 86"
];

window.matchingGameHardElements2 = [

    "thirteen",
    "ninetyNine",
    "twentyFour"
];


window.matchingGameHardProblems3 = [

    "65 - 24",
    "13 * 3",
    "44 + 38"
];

window.matchingGameHardElements3 = [

    "fortyOne",
    "thirtyNine",
    "eightyTwo"
];
